src/Time.svelte

```diff
@@ -63,6 +65,30 @@
     );
   $: multiplier = isMinuteView ? 5 : 1;
 
+  $: startDateHour = startDate && new Date(startDate).getHours();
+  $: startDateMinutes = startDate && new Date(startDate).getMinutes();
+  $: endDateHour = endDate && new Date(endDate).getHours();
+  $: endDateMinutes = endDate && new Date(endDate).getMinutes();
+  
+  $: isHourDisabled = (hour) => {

```

Comment:
You should always compare if `startDate` or `endDate` are on the same day as `date`